### ISSUE TITLE
JS clients not connecting after update to 1.6

### ISSUE THREAD

**Issue by voolik** [vernemq/vernemq#950]
### Environment

- VerneMQ Version: 1.6.1
- OS: Debian Jessie (hosted in Docker/Kubernetes)
- Erlang/OTP version (if building from source): Prebuilt binary
- VerneMQ configuration (vernemq.conf) or the changes from the default
No extraordinary changes. TLS, client certificate authentication.

### Expected behavior

- All clients should be able to connect to VerneMQ without issues.

### Actual behaviour

- Clients using C++ library connect fine. Clients using Node.js MQTT library fail to connect.
VerneMQ error log gives these error messages:
```
2018-11-09 15:53:17.103 [error] <0.508.0>@vmq_mqtt_pre_init:consume_var_header:86 Ranch listener {{172,17,0,2},8883} terminated with reason: no function clause matching vmq_mqtt_pre_init:consume_var_header(<<>>) line 86
2018-11-09 15:53:18.132 [error] <0.511.0>@vmq_mqtt_pre_init:consume_var_header:86 CRASH REPORT Process <0.511.0> with 0 neighbours crashed with reason: no function clause matching vmq_mqtt_pre_init:consume_var_header(<<>>) line 86
```

Everything works fine on version 1.5.0 with no configuration changes.

**Comment by ioolkos** [comment]
@voolik thanks. You're using MQTT.js? can you give us some example how you connect from it? (to see if we can reproduce things)

**Comment by voolik** [comment]
@ioolkos Correct, latest version (2.18.8).

```
const mqttUrl = process.env.MQTT_URL || "mqtt://127.0.0.1:1883";

let mqttClient: mqtt.MqttClient;

if (mqttUrl.indexOf("mqtts") === 0) {
  mqttClient = mqtt.connect(mqttUrl, {
    ca: fs.readFileSync("ca.crt"),
    cert: fs.readFileSync("client.crt"),
    key: fs.readFileSync("client.key"),
    clientId: hostname
  });
} else {
  logger.debug("No TLS configured for MQTT.");
  mqttClient = mqtt.connect(mqttUrl);
}

mqttClient.on("connect", () => {
  logger.info("mqttClient: connected.");
  mqttClient.subscribe("status/#");
});
```

It never reaches the on connect section (does not log). When you shut down VerneMQ, it says that it was disconnected before TLS handshake was finished.

**Comment by larshesel** [comment]
Hi, can you reproduce this without TLS? The TLS handshake is done before any MQTT data is exchanged and it would be good to make sure this isn't really a TLS issue. Also without TLS it would be easier to dump and understand the network traffic with wireshark or tcpdump

**Comment by larshesel** [comment]
ping @voolik ?

**Comment by voolik** [comment]
Apologies for the delay, I have been a little busy. I will test it out this week and let you know of the results.

**Comment by larshesel** [comment]
When everything worked fine with 1.5.0 was that with another version of the MQTT.js library? Can you retest with the same version (2.18.8) and see if this works on 1.5.0?

**Comment by larshesel** [comment]
I believe we figured out what the problem was and we've created a fix for this in #970

**Comment by voolik** [comment]
> When everything worked fine with 1.5.0 was that with another version of the MQTT.js library? Can you retest with the same version (2.18.8) and see if this works on 1.5.0?

@larshesel Same version of MQTT.js library was used in all cases, 2.18.8.

I have now completed testing without TLS - the issue does occur only when TLS is enabled. When connecting without TLS (port 1883), JS clients can connect successfully.
